src/devices/rpLidar2/sdk_original/sdk/src/rplidar_driver.cpp

```diff
@@ -0,0 +1,1026 @@
+/*
+ *  RPLIDAR SDK
+ *
+ *  Copyright (c) 2009 - 2014 RoboPeak Team
+ *  http://www.robopeak.com
+ *  Copyright (c) 2014 - 2016 Shanghai Slamtec Co., Ltd.
+ *  http://www.slamtec.com
+ *
+ */
+/*
+ * Redistribution and use in source and binary forms, with or without 
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, 
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, 
+ *    this list of conditions and the following disclaimer in the documentation 
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, 
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR 
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, 
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, 
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; 
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR 
+ * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, 
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include "sdkcommon.h"
+#include "hal/abs_rxtx.h"
+#include "hal/thread.h"
+#include "hal/locker.h"
+#include "hal/event.h"
+#include "rplidar_driver_serial.h"
+
+#ifndef min
+#define min(a,b)            (((a) < (b)) ? (a) : (b))
+#endif
+
+namespace rp { namespace standalone{ namespace rplidar {
+
+
+// Factory Impl
+RPlidarDriver * RPlidarDriver::CreateDriver(_u32 drivertype)
+{
+    switch (drivertype) {
+    case DRIVER_TYPE_SERIALPORT:
+        return new RPlidarDriverSerialImpl();
+    default:
+        return NULL;
+    }
+}
+
+
+void RPlidarDriver::DisposeDriver(RPlidarDriver * drv)
+{
+    delete drv;
+}
+
+
+
+// Serial Driver Impl
+
+RPlidarDriverSerialImpl::RPlidarDriverSerialImpl() 
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Member variable 'RPlidarDriverSerialImpl::_is_previous_capsuledataRdy' is not initialized in the constructor.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5162834)